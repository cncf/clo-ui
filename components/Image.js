module.exports = require('../dist/cjs/components/Image.js');
