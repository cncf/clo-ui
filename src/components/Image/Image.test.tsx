import { render, screen } from '@testing-library/react';

import { Image } from './Image';

const defaultProps = {
  url: 'http://img.url',
  dark_url: 'http://dark-img.url',
  alt: 'image',
  effective_theme: 'light',
};

describe('Image', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Image {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<Image {...defaultProps} />);

    const img = screen.getByAltText('image');
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'http://img.url');
  });

  it('renders dark image', () => {
    render(<Image {...defaultProps} effective_theme="dark" />);

    const img = screen.getByAltText('image');
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'http://dark-img.url');
  });

  it('renders placeholder', () => {
    render(<Image {...defaultProps} url={null} />);

    expect(screen.queryByAltText('image')).toBeNull();
    expect(screen.getByTestId('img-placeholder')).toBeInTheDocument();
  });
});
