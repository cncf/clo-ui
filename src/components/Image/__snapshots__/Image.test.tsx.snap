// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Image creates snapshot 1`] = `
<DocumentFragment>
  <img
    alt="image"
    aria-hidden="true"
    src="http://img.url"
  />
</DocumentFragment>
`;
